### PR TITLE
LSV and Postfinance Direct Debit file generation

### PR DESCRIPTION
**Pull request by CSester**
This pull request introduces 2 new features for swiss localization: LSV and Postifnance Direct Debit file generation.
The module uses the "Direct debit order" module (https://github.com/OCA/banking/tree/7.0/account_direct_debit) and the "Banking mandate" module (https://github.com/CompassionCH/banking/tree/7.0-banking-mandates/account_banking_mandate).
The module also uses the changes made in pull request OCA/l10n-switzerland#34 to generate ESR reference if an invoice's associate bank account is LSV-configured.

As the module depends on the banking mandate, which is a new feature in banking-addons, this pull request should not be merged before OCA/banking#27 is merged.

### CODE REVIEW

**Comment by nbessi** [comment]
Thanks for the PR. Some prerequisit seems to be missing in travis file

**Comment by CSester** [comment]
Yes, I put the 7.0 branch of the OCA/banking module as prerequist. This way, the Travis CI build will only success once the banking-mandate is integrated in the banking-7.0 branch.

**Comment by nbessi** [comment]
- [x] On hold of https://github.com/OCA/banking/pull/27

**Comment by CSester** [comment]
Ok, let's work on this. OCA/banking#27 has just been merged.

**Comment by CSester** [comment]
About Travis:
- test-server seems to fail because of a warning comming from "credit.control.dunning.fees.computer" module (external)
- Pylint tells that some code is unreachable. These are validation callbacks called from _constraints. Is there a clean way to solve this "bug"? Or should I do some dummy call to these 2 functions ?

**Comment by nbessi** [comment]
@CSester I should fix credit.control.dunning.fees.computer

**Comment by CSester** [comment]
@nbessi And what about Pylint?

File: l10n_ch_lsv_dd/model/bank.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        """ Check if given LSV Identifier is valid """
+        if not isinstance(lsv_identifier, basestring):
+            return False
+        try:
+            lsv_identifier.decode('ascii')
+        except UnicodeDecodeError:
+            raise orm.except_orm('ValidateError',
+                                 _('LSV identifier should contain only ASCII'
+                                   'caracters.'))
+            return False
```
Comment (nbessi) [inline l10n_ch_lsv_dd/model/bank.py]: pylint shows a indentation error here

File: l10n_ch_lsv_dd/model/bank.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        """ Check if given Postfinance DD Identifier is valid """
+        if not isinstance(dd_identifier, basestring):
+            return False
+        try:
+            dd_identifier.decode('ascii')
+        except UnicodeDecodeError:
+            raise orm.except_orm('ValidateError',
+                                 _('DD identifier should contain only ASCII '
+                                   'caracters.'))
+            return False
```
Comment (nbessi) [inline l10n_ch_lsv_dd/model/bank.py]: same here

**Comment by davidwul** [comment]
@nbessi where you able to look at credit.control.dunning.fees ? Anything else blocking the PR?

**Comment by nbessi** [comment]
Hello,

Probably this Friday.

Regards

Nicolas

2014-10-21 16:36 GMT+02:00 davidwul notifications@github.com:

> @nbessi https://github.com/nbessi where you able to look at
> credit.control.dunning.fees ? Anything else blocking the PR?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/l10n-switzerland/pull/37#issuecomment-59937365.

## 

Nicolas Bessi
Senior ERP consultant
Business Solution technical manager

Camptocamp SA
PSE A
CH-1015 Lausanne

http://openerp.camptocamp.com

Direct:   +41 21 619 10 26
Office:   +41 21 619 10 10

**Comment by nbessi** [comment]
@CSester  the PR is here https://github.com/OCA/account-financial-tools/pull/60

**Comment by nbessi** [comment]
Nice it is green :+1:

**Comment by ecino** [comment]
Rebased on branch 7.0 to integrate latest changes. :+1:

**Comment by nbessi** [comment]
:+1:

**Pull request merged**